sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -221,6 +221,15 @@ private[spark] object HiveUtils extends Logging {
       .booleanConf
       .createWithDefault(true)
 
+  val QUOTE_HIVE_STRUCT_FIELD_NAME =
+    buildConf("spark.sql.hive.quoteHiveStructFieldName")
+      .doc("When true, for a column defined in struct type, when it contains special characters " +
+        "in the field name, Spark will quote it for verification. E.g. struct<x:int,y.z:int>" +
+        " is read as struct<`x`:int,`y.z`:int> for verification.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
does it mean we disable the feature for the read side as well? I think we only need to disable it for the write side.

cc @yaooqinn

Reply:
I guess you still misunderstand some here, we don't even apply reg-replace for the write-side

Reply:
I think there are two things here:
1. support reading Hive tables with special chars in the column name
2. support creating hive-compatible tables with special chars in the column name

1 is always safe to do, but 2 need a flag because it may break compatibility.